src/promql/src/extension_plan/series_divide.rs

```diff
@@ -255,18 +255,20 @@ impl Stream for SeriesDivideStream {
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
         loop {
-            if let Some(batch) = self.buffer.take() {
+            if let Some(batch) = self.buffer.clone() {
```

Comment:
```suggestion
            // It has to be cloned here, otherwise the later ready! will mess things up
            if let Some(batch) = self.buffer.clone() {
```